include/versioningCompiler/CompilerImpl/ClangLLVM/OptUtils.hpp

```diff
@@ -282,9 +288,11 @@ static void AddOptimizationPasses(llvm::legacy::PassManagerBase &MPM,
   if (TM) {
     TM->adjustPassManager(Builder);
   }
+  #if LLVM_MAJOR_VERSION <15
   if (Coroutines) {
-    llvm::addCoroutinePassesToExtensionPoints(Builder);
+    llvm::addCoroutinePassesToExtensionPoints(Builder); // todo No clue on how to replace this for LLVM-15
   }
+  #endif
```

Comment:
this was a sort of clone from mainstream `opt`. What do they do there?

Reply:
I'd guess it was a sort of a clone from [this](https://github.com/llvm/llvm-project/blob/release/14.x/llvm/tools/opt/opt.cpp), right?
This changes quite a lot between  [14](https://github.com/llvm/llvm-project/blob/release/14.x/llvm/tools/opt/opt.cpp) and  [15](https://github.com/llvm/llvm-project/blob/release/15.x/llvm/tools/opt/opt.cpp). I guess it would not be retrocompatible anyway.

Reply:
Well, yes. Ten versions earlier, but yes.

Retrocompatibility of this class is probably hard to obtain.
Does the 14 version work well when compiled against LLVM 15?

You could test it by scheduling a generic optimisation pass, say `mem2reg` in a test case.
If that does not work, I suggest to selectively disable this Compiler Implementation from cmake, depending on the version of LLVM.

Reply:
Well, [Test.cpp](https://github.com/skeru/libVersioningCompiler/pull/3/files#diff-bce024b33fc3cc08c6ad47c2fca5f2a12154e381a0c4345c94dc611c9543721dR115) seems to schedule an optimization pass and both tests (libVC_test and libVC_testJit) succeed on llvm 15. Can it be considered as working?

Reply:
I would call it a fair test.

However, can you please investigate was was saying the commit which removed these lines in mainstream LLVM?
I suspect that it was integrated into something new, but just to be sure...

Reply:
That's a good point, I'll look into that!

Reply:
Updated! It is properly working for LLVM 13, 14 and 15!